### PR TITLE
zeiler2012adadelta_1 initial summary

### PR DESCRIPTION
**Pull request by sritank**
Requesting review from @mattkelleher and @zhaoyue-zephyrus

### CODE REVIEW

File: _summaries/zeiler2012adadelta_1.md [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
-paper: {{paper_tag}}
+paper: {{zeiler2012adadelta}}
 # Please fill out info below
-author: # Your GitHub id
-score: # How did you like this paper 0(dislike) to 10(love)
+author:  sritank
+score:  10
 ---
 
 TODO: Summarize the paper:
```
Comment (mattkelleher) [inline _summaries/zeiler2012adadelta_1.md:10]: Small personal preference for formatting, can remove the "TODO summarize the paper" as well as the other bullets for the template "What is the core idea?" etc etc and replace with more insightful titles/headers. (eg. Main Idea,  Implementation, etc etc)

File: _summaries/zeiler2012adadelta_1.md [hunk 2]
```diff
@@ -1,19 +1,48 @@
...
+
 * What is the core idea?
+- Adaptively change the learning rate in a way that it reduces oscillations i.e. slows down close to minimas, and is large otherwise.
+- Change step size based on the windowed accumulated gradient and also accumulated step sizes, which only require 1st order computations.
+- Also scale the step size by the accumulated previous step sizes, giving it the correct units as if it were a 2nd order adaptive method
```
Comment (mattkelleher) [inline _summaries/zeiler2012adadelta_1.md:15]: might be interesting to note that ADADELTA is a variation of ADAGRAD that address' the following 2 issues with ADAGRAD: (directly from the papaer) 
"1.the continual decay of learning rates throughout training, and 2) the need for a manually selected global learning rate."
EDIT: I see that you mentioned ADAGRAD below but might be good to bring up the 2 issues ADADELTA solves ADAGRAD does not (either verbally during presentation or in text)

File: _summaries/zeiler2012adadelta_1.md [hunk 3]
```diff
@@ -1,19 +1,48 @@
...
-score: # How did you like this paper 0(dislike) to 10(love)
+author:  sritank
+score:  10
 ---
 
 TODO: Summarize the paper:
+
 * What is the core idea?
+- Adaptively change the learning rate in a way that it reduces oscillations i.e. slows down close to minimas, and is large otherwise.
+- Change step size based on the windowed accumulated gradient and also accumulated step sizes, which only require 1st order computations.
```
Comment (mattkelleher) [inline _summaries/zeiler2012adadelta_1.md:14]: may not need to add but would be good to mention that this windowed accumulation also makes sure that updates after many rounds of training are impactful and not "drowned out" by the massive accumulated denominator from ALL previous steps (as in ADAGRAD)

File: _summaries/zeiler2012adadelta_1.md [hunk 4]
```diff
@@ -1,19 +1,48 @@
...
+![ADADELTA_denom](exponential_decay_average.png)
+where epsilon is used to condition the denominator for numerical purposes.
+- By assuming locally smooth curvature, they scaled up the step size using the exponentially decaying average of previous step sizes.
+
+![ADADELTA_step](ADADELTA_step.png)
 * How well does the paper perform?
-* What interesting variants are explored?
+- numerator RMS term lags behind the denominator by 1 time step, making it robust to large sudden gradients. Denominator increases slowing down the progress before numerator can blow up.
+- The method makes approximations on Hessian and local curvature, giving it second order characteristics while still costing only one gradient computation per iteration.
+- Training method is used on DNNs with sigmoid and ReLU units. The DNNs were trained for MNIST classification and speech recognition.
```
Comment (mattkelleher) [inline _summaries/zeiler2012adadelta_1.md:34]: Was the training only on DNNs with sigmoid and ReLU by design? Would you expect ADADELTAto work / not work on DNN's with different non-linear activations?

**Review by mattkelleher — COMMENTED** [review]
Overall looks like a really solid review! I think the figures are interesting/expressive and you cover the main features of ADADELTA

File: _summaries/zeiler2012adadelta_1.md (hunk 1 above)
Comment (sritank) [reply _summaries/zeiler2012adadelta_1.md:10]: Done

File: _summaries/zeiler2012adadelta_1.md (hunk 3 above)
Comment (sritank) [reply _summaries/zeiler2012adadelta_1.md:14]: Added it

File: _summaries/zeiler2012adadelta_1.md (hunk 2 above)
Comment (sritank) [reply _summaries/zeiler2012adadelta_1.md]: Added it to draft

File: _summaries/zeiler2012adadelta_1.md (hunk 4 above)
Comment (sritank) [reply _summaries/zeiler2012adadelta_1.md]: I believe the method is independent of the activation function. As long as we can calculate the gradient at each time step, the adaptive learning rate can be calculated at any point in the training. 
In their experiments they only seem to have tried sigmoid and ReLU
